### PR TITLE
[MINOR][TEST]Correct some unit test mistakes

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?

Correct some unit test mistakes.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by 10110346** [comment]
> 
> 
> Shall we fix those when we actually though the codes around there? I don't think those cleanups are encouraged in general.

I've already checked the codes around there.
I think such mistakes should be corrected now:
https://github.com/apache/spark/blob/1c3e57abd4a0cda5981e08ee53fe8fa2983f2943/sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala#L478

File: sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala [hunk 1]
```diff
@@ -170,7 +170,7 @@ class InsertSuite extends DataSourceTest with SharedSQLContext {
 
     // Writing the table to more part files.
     val rdd2 = sparkContext.parallelize((1 to 10).map(i => s"""{"a":$i, "b":"str$i"}"""), 10)
-    spark.read.json(rdd1.toDS()).createOrReplaceTempView("jt2")
+    spark.read.json(rdd2.toDS()).createOrReplaceTempView("jt2")
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala]: this one, yea okay.

File: sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala [hunk 2]
```diff
@@ -476,7 +475,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
 
     val keyValue1 = ds.groupByKey(t => (t._1, "key")).mapValues(t => (t._2, "value"))
     val agged1 = keyValue1.mapGroups { case (g, iter) => (g._1, iter.map(_._1).sum) }
-    checkDataset(agged, ("a", 30), ("b", 3), ("c", 1))
+    checkDataset(agged1, ("a", 30), ("b", 3), ("c", 1))
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala]: this one, yup okay.

**Comment by HyukjinKwon** [comment]
Let's revert other changes. You can actually suggest such change right away in Github as well when you review other PRs.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 3]
```diff
@@ -308,7 +308,6 @@ private[spark] class SparkSubmit extends Logging {
     val isMesosCluster = clusterManager == MESOS && deployMode == CLUSTER
     val isStandAloneCluster = clusterManager == STANDALONE && deployMode == CLUSTER
     val isKubernetesCluster = clusterManager == KUBERNETES && deployMode == CLUSTER
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]:
```suggestion
```

Like this.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLog.scala [hunk 4]
```diff
@@ -56,7 +56,6 @@ class OffsetSeqLog(sparkSession: SparkSession, path: String)
       throw new IllegalStateException("Incomplete log file")
     }
 
-    val version = parseVersion(lines.next(), OffsetSeqLog.VERSION)
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLog.scala]: The problem here is that this removes a call to lines.next(), which has a side effect. I think you can only remove `val version =`. Same above

**Review by srowen — CHANGES_REQUESTED** [review]

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 3 above)
Comment (10110346) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: OK, thank you

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLog.scala (hunk 4 above)
Comment (10110346) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLog.scala]: Thanks

**Comment by 10110346** [comment]
retest this please

**Review by srowen — APPROVED** [review]
Yes these fixes are OK. I don't mind removing some dead code here and there too but we can just do this part now.

**Comment by srowen** [comment]
Merged to master

**Comment by 10110346** [comment]
> 
> 
> Yes these fixes are OK. I don't mind removing some dead code here and there too but we can just do this part now.

@srowen  Thanks,  should I create a PR to remove some  dead code now?

**Comment by srowen** [comment]
I wouldn't worry about it right now unless there is some minor performance implication to removing the unused declarations, but if there is, sure.

**Comment by 10110346** [comment]
Okay, it doesn't seem too urgent.

**Pull request closed without merging**